### PR TITLE
refactor: migrate test_benchmarks to null safety

### PR DESCRIPTION
**Pull request by amanv8060**
Migrate test_benchmarks to null safety

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: test_benchmarks/benchmarks/scroll.dart [hunk 1]
```diff
@@ -34,15 +32,15 @@ const _minFreeRoomRequirement = 5.0;
 
 /// Whether [small] is a subset of [large] and has sufficient room
 /// inside [large], at the end of [large] specified by [axisDirection].
-bool _hasSufficientFreeRoom({
-  @required Rect large,
-  @required Rect small,
-  @required AxisDirection axisDirection,
+bool? _hasSufficientFreeRoom({
```
Comment (mit-mit) [inline test_benchmarks/benchmarks/scroll.dart:35]: It doesn't look like the call to this handles it potentially being null, so I'd make it non-nullable, and then in line 43 below give it an initial value (either true or false) before the switch statement.

**Comment by amanv8060** [comment]
Can't figure out the failure, meanwhile closing it for anyone else to have a look.

**Comment by guidezpl** [comment]
Re-opening to see/fix the failures

**Comment by mit-mit** [comment]
@guidezpl I can't make sense of this error. Do you understand what's going on?
https://github.com/flutter/gallery/runs/8087094487?check_suite_focus=true

**Comment by pennzht** [comment]
> @guidezpl I can't make sense of this error. Do you understand what's going on? https://github.com/flutter/gallery/runs/8087094487?check_suite_focus=true

Taking a look now.

**Comment by pennzht** [comment]
Thank you for your message!

I tested the web benchmarks a few times locally, and it worked every time after updating the package versions.

A similar PR https://github.com/flutter/gallery/pull/768 passes all checks.

**Have you tried re-running the web benchmark checks?** Does the failure persist?
It could be possible that this test is flaky, although I haven't observed any flakiness before.

**Comment by mit-mit** [comment]
I've tried several re-runs over several days. I just did it again, this time selecting to include debug info. But I still don't see anything that helps me understand what's going on:
https://github.com/flutter/gallery/runs/8191884582?check_suite_focus=true

**Review by guidezpl — APPROVED** [review]
Nice. I'll fix the goldens issue separately (the banner divider changed color) after the change is rolled in

**Pull request merged**